RxJavaTemplate/config/jacoco.gradle

```diff
@@ -44,13 +44,37 @@ task jacocoTestReport(type: JacocoReport) {
         "$project.rootDir/data/src/main/java"
     ])
 
+    /**
+     * Once enabled [testCoverageEnabled], Jacoco will capture the coverage and store them in
+     * [${project.module}/jacoco.exec]. We need to add all [jacoco.exec] to here.
+     * [${project.module}/build/jacoco/testFlavorDebugTest.exec] won't have the result anymore, so we
+     * can safety get rid of them.
+     * Reference: https://stackoverflow.com/a/67626100/5187859
+     * Issue tracker 1: https://issuetracker.google.com/issues/171125857#comment20
+     * Issue tracker 2: https://issuetracker.google.com/issues/195860510
+     */
     executionData.from = fileTree(dir: project.rootDir, includes: [
-        "app/build/jacoco/testStagingDebugUnitTest.exec",
-        "data/build/jacoco/testStagingDebugUnitTest.exec"
+        "app/jacoco.exec",
+        "common/jacoco.exec"
     ])
 
 }
 
+/**
+ * Workaround to bypass "Caused by: java.lang.IllegalStateException:
+ * Cannot process instrumented class...
+ * Please supply original non-instrumented classes." issue.
+ *
+ * Application projects that depend on variants of libraries that have test coverage enabled will
+ * still not work as app code should be instrumented on the fly, while library code should not be.
+ * https://issuetracker.google.com/issues/171125857#comment26
+ */
+tasks.withType(Test.class).configureEach {
+    jacoco {
+        includes = ["com.application.*"] // include only application classes
```

Comment:
@doannimble As this is a workaround, we don't know exactly the reason behind it. I tried to verify what we need to provide in the `includes` field but actually, it could be **anything** lol. More feedback is welcome please 🙌 

- With `com.application.*`

![image](https://user-images.githubusercontent.com/16315358/162172780-f092dd4f-b3fe-49ae-a4b1-e0d93433c8ee.png)


- With `com.abc.*`

![image](https://user-images.githubusercontent.com/16315358/162172584-41853757-6199-4b21-a34a-b1544a133e8f.png)


- Without

![image](https://user-images.githubusercontent.com/16315358/162170412-54ccea78-32b9-4d8c-83ff-355085748766.png)